controllers/argocd/applicationset.go

```diff
@@ -206,12 +216,19 @@ func (r *ReconcileArgoCD) reconcileApplicationSetDeployment(cr *argoprojv1a1.Arg
 	if err := controllerutil.SetControllerReference(cr, deploy, r.Scheme); err != nil {
 		return err
 	}
```

Comment:
We probably should do that also only if ApplicationSet is not disabled. It just makes no sense to call `SetControllerReference` if we're not going to reconcile anyway.